secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/VaultRetryService.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.cloudbreak.service.secret.service;
+
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.metrics.MetricService;
+import com.sequenceiq.cloudbreak.common.metrics.type.MetricType;
+import com.sequenceiq.cloudbreak.service.Retry;
+
+@Service
+public class VaultRetryService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VaultRetryService.class);
+
+    private final MetricService metricService;
+
+    public VaultRetryService(MetricService metricService) {
+        this.metricService = metricService;
+    }
+
+    @Retryable(
+            value = Retry.ActionWentFailException.class,
+            maxAttemptsExpression = "${vault.retry.maxattempt:5}",
+            backoff = @Backoff(delayExpression = "${vault.retry.delay:2000}",
+                    multiplierExpression = "${vault.retry.multiplier:2}",
+                    maxDelayExpression = "${vault.retry.maxdelay:10000}")
+    )
+    public <T> T testVaultRead(Supplier<T> action) throws Retry.ActionWentFailException {
```

Comment:
Why `test`? Simply calling it `read` and other method `write` would be more straightforward in my opinion. The service name already contains the information that it retries every operation. I don't understand why `test` is applicable here.

Reply:
ok